src/libraries/System.Private.Xml/src/System/Xml/Xsl/Xslt/QilGenerator.cs

```diff
@@ -196,13 +196,14 @@ private QilExpression Compile(Compiler compiler)
             }
 
             // Create list of all early bound objects
-            Dictionary<string, Type?> scriptClasses = compiler.Scripts.ScriptClasses;
+            Scripts.TrimSafeDictionary scriptClasses = compiler.Scripts.ScriptClasses;
             List<EarlyBoundInfo> ebTypes = new List<EarlyBoundInfo>(scriptClasses.Count);
-            foreach (KeyValuePair<string, Type?> pair in scriptClasses)
+            foreach (string key in scriptClasses.Keys)
```

Comment:
Why the change here to enumerate the keys and then index into the dictionary? Why not just expose `GetEnumerator` from `TrimSafeDictionary`?

Reply:
The problem here is that if we keep it as it was, then we would get a warning (even when using TrimmerSafeDictionary) since the Type would be comming form method: `System.Collections.Generic.KeyValuePair<Tkey, TValue>.get` which wouldn't be annotated. Of course we could suppress that, but by doing it the way I'm doing it here, the value gets accessed via the index operator on TrimmerSafeDictionary, so that way the linker will not throw a warning and we won't need an additional suppression.